### ISSUE TITLE
Unsafe inline cast from Tensor<2,dim> to SymmetricTensor<2,dim> in v8.3.0

### ISSUE THREAD

**Issue by jppelteret** [dealii/dealii#1839]
When one adds a `SymmetricTensor` to a normal `Tensor`, the `Tensor` is preferrentially cast to a `SymmetricTensor` leading to an unexpected result:
```C++
const Tensor<2,dim> F_q_point = unit_symmetric_tensor<dim>() + Grad_u[q_point];
```
This can (and must) of course be avoided in the following manner
```C++
const Tensor<2,dim> F_q_point = static_cast< Tensor<2,dim> >(unit_symmetric_tensor<dim>()) + Grad_u[q_point];
```
and the problem is detected at run time in debug mode
```
    dealii::SymmetricTensor<2, 3, double>::SymmetricTensor(const Tensor<2, dim, Number> &)
The violated condition was:
    t[0][1] == t[1][0]
The name and call sequence of the exception was:
    ExcInternalError()
```

However, it would be nice to force a cast the other way round, perhaps by defining `operator+` for these cases.

**Comment by tamiko** [comment]
> When one adds a `SymmetricTensor` to a normal `Tensor`, the `Tensor`
> is preferrentially cast to a `SymmetricTensor` leading to an
> unexpected result:
> ```C++
> const Tensor<2,dim> F_q_point = unit_symmetric_tensor<dim>() + Grad_u[q_point];
> ```
> This can (and must) of course be avoided in the following manner
> ```C++
> const Tensor<2,dim> F_q_point = static_cast< Tensor<2,dim>

> [...]

> However, it would be nice to force a cast the other way round, perhaps
> by defining `operator+` for these cases.

This is a bit surprising because I already introduced the correct
overloads in

  commit d86d56b2006c03f7f40f94dd96fb4363cf67d28e
  Author: Matthias Maier <tamiko@43-1.org>
  Date:   Wed Oct 28 22:47:29 2015 -0500

and in fact the following code works as expected:

```
#include <deal.II/base/tensor.h>
#include <deal.II/base/symmetric_tensor.h>

using namespace dealii;

int main ()
{
  Tensor<2,2> Grad_u;
  Grad_u[0][0] = 1.;
  Grad_u[0][1] = 2.;
  Grad_u[1][0] = 3.;
  Grad_u[1][1] = 4.;

  const auto F_q_point = unit_symmetric_tensor<2>() + Grad_u;

}
```

**Comment by tamiko** [comment]
Ah, sorry. :-D

I missed the "in 8.3.0".

**Comment by tamiko** [comment]
Fixed in d86d56b ;-)

**Comment by jppelteret** [comment]
Awesome. Sorry I missed that @tamiko . Its sometimes hard to keep up with all the fixes coming in :-)
